specifyweb/frontend/js_src/lib/components/QueryComboBox/helpers.ts

```diff
@@ -230,3 +230,58 @@ export function pendingValueToResource(
     typeof fieldName === 'string' ? { [fieldName]: pendingValue } : {}
   );
 }
+
+const DEFAULT_RECORD_PRESETS = {
+  CURRENT_AGENT: () => userInformation.agent.resource_uri,
+  CURRENT_USER: () => userInformation.resource_uri,
+  BLANK: () => null,
+} as const;
+type DefaultRecordPreset = keyof typeof DEFAULT_RECORD_PRESETS;
+
+export function useQueryComboBoxDefaults({
+  resource,
+  field,
+  defaultRecord,
+}: {
+  readonly resource: SpecifyResource<AnySchema> | undefined;
+  readonly field: Relationship;
+  readonly defaultRecord?: string | undefined;
+}): void {
+  if (resource === undefined || !resource.isNew()) return;
+
+  if (defaultRecord !== undefined) {
+    const defaultUri: string | null =
+      defaultRecord in DEFAULT_RECORD_PRESETS
+        ? DEFAULT_RECORD_PRESETS[defaultRecord as DefaultRecordPreset]()
+        : defaultRecord;
+
+    resource.set(field.name, resource.get(field.name) ?? defaultUri, {
+      silent: true,
+    });
+    // The following cases need to be kept for outdated forms that do not use the defaultRecord property.
+  } else if (field.name === 'cataloger') {
+    const record = toTable(resource, 'CollectionObject');
+    record?.set(
+      'cataloger',
+      record?.get('cataloger') ?? userInformation.agent.resource_uri,
+      {
+        silent: true,
+      }
+    );
+  } else if (field.name === 'specifyUser') {
+    const record = toTable(resource, 'RecordSet');
+    record?.set(
+      'specifyUser',
+      record?.get('specifyUser') ?? userInformation.resource_uri
+    );
+  } else if (field.name === 'receivedBy') {
+    const record = toTable(resource, 'LoanReturnPreparation');
+    record?.set(
+      'receivedBy',
+      record?.get('receivedBy') ?? userInformation.agent.resource_uri,
+      {
+        silent: true,
+      }
+    );
+  }
```

Comment:
This suggestion might be hyper-optimization so I'll leave it up to you to decide whether to include it or not:
These fallback else-if blocks could potentially be wrapped inside a function `setLegacyDefaults` or something similar just to make the hook more readable. The hook would essentially look like:

```
if (defaultRecord !== undefined) {
    // do something
} else {
   // setLegacyDefaults()
}
```